aspnetcore/mvc/razor-pages/index.md

```diff
@@ -69,398 +67,254 @@ The associations of URL paths to pages are determined by the page's location in
 
-### Using multiple handlers
+## TempData
 
-Let's update this form to support multiple operations. A visitor to the site can either join the mailing list or ask for a free quote.
+The `[TempData]` attribute is new in 2.0.0 and is supported on controllers and pages. In 2.0.0, the default storage for temp data is cookies. A session provider is no longer required by default.
 
-If you want one page to handle multiple logical actions, you can use *named handler methods*. Any text in the name after `On<Verb>` and before `Async` (if present) in the method name is considered a handler name. The handler methods in the following example have the handler names `JoinMailingList` and `RequestQuote`:
+TODO provide sample moving temp data bewteen pages.
 
-*MyApp/Pages/Contact.cshtml*
+<a name="mhpp"></a>
+##  Multiple handlers per page
 
-```html
-@page
-@inject ApplicationDbContext Db
+The following page generates markup for two page handlers using the `asp-page-handler` tag helper:
 
-@functions {
+[!code-html[main](index/sample/RazorPagesContacts2/Pages/Customers/CreateFATH.cshtml?highlight=12-13)]
 
-    [BindProperty]
-    public Contact Contact { get; set; }
-    
-    public async Task<IActionResult> OnPostJoinMailingListAsync()
-    {
-        ...
-    }
+<!-- Review: Where is the FormActionTagHelper?  How is it used in conjunction? -->
+The form in the preceding example has two submit buttons, each using the new `FormActionTagHelper` in conjunction to submit to a different URL. The `asp-handler` attribute is a companion to `asp-page` and generates URLs that submit to each of the handler methods defined by the page. We don't need to specify `asp-page` because we're linking to the current page.
 
-    public async Task<IActionResult> OnPostRequestQuoteAsync()
-    {
-        ...
-    }
-}
-<div class="row">
-    <div class="col-md-3">
-        <p>Enter your contact info here we will email you about our fine products! Or get a free quote!</p>
-        <div asp-validation-summary="All"></div>
-        <form method="POST">
-            <div>Name: <input asp-for="Contact.Name" /></div>
-            <div>Email: <input asp-for="Contact.Email" /></div>
-            <input type="submit" asp-page-handler="JoinMailingList" value="Join our mailing list"/>
-            <input type="submit" asp-page-handler="RequestQuote" value="Get a free quote"/>
-        </form>
-    </div>
-</div>
-```
+The code-behind file:
+
+[!code-cs[main](index/sample/RazorPagesContacts2/Pages/Customers/CreateFATH.cshtml.cs?highlight=20,32)]
+
+The preceding code uses *named handler methods*. Named handler methods are created by taking the text in the name after `On<HTTP Verb>` and before `Async` (if present). In the preceding example, the page methods are OnPost**JoinList**Async and OnPost**JoinListUC**Async. The handler names when *OnPost* and *Async* are removed are `JoinList` and `JoinListUC`.
```

Comment:
~~The handler names when *OnPost* and *Async* are removed are `JoinList` and `JoinListUC`.~~
With *OnPost* and *Async* removed, the handler names are `JoinList` and `JoinListUC`.